app/models/qernel/node_api/cost.rb

```diff
@@ -435,6 +435,21 @@ def variable_costs_per_typical_input(include_waste: true)
       #
       # Returns the the yearly fuel costs for one single plant.
       def fuel_costs
+        fetch(:fuel_costs) do
+          if typical_input && typical_input < 0
+            raise IllegalNegativeError.new(self, :typical_input, typical_input)
+          end
+          typical_input * weighted_carrier_cost_per_mj
+        end
+      end
+
+      # Internal: Calculates the fuel costs for a single dispatchable plant
```

Comment:
"Dispatchable plant" is not entirely the correct phrasing since the method is used for flexible electricity consumers, but also for electricity storage technologies. I would say it calculates the fuel costs for one single plant, with the electricity costs calculated on an hourly basis. This ties into my comment about the name of the method itself.